src/pudl_archiver/archivers/validate.py

```diff
@@ -0,0 +1,35 @@
+"""Defines models used for validating/summarizing an archiver run."""
+from typing import Literal
+
+from pydantic import BaseModel
+
+
+class ValidationTestResult(BaseModel):
+    """Class containing results of a validation test, and metdata about the test."""
+
+    name: str
+    description: str
+    ignore_failure: bool = False
+    resource_name: str | None = None
+    result: bool
+    note: str | None = None
```

Comment:
What's the difference between `description` and `note`?

Reply:
The idea was that `note` was for adding details like why the test failed. I could change to something like `error_message` to be more clear, but I wasn't sure if I should constrain it to only be for error messages, or leave the option open to include other types of notes.